rdr_service/dao/site_hierarchy_dao.py

```diff
@@ -0,0 +1,83 @@
+from typing import Dict
+
+from flask import jsonify, Response
+from sqlalchemy import or_
+from sqlalchemy.orm import Session, query
+from werkzeug.exceptions import BadRequest
+
+from rdr_service.dao.base_dao import BaseDao
+from rdr_service.model.site import Site
+from rdr_service.model.organization import Organization
+from rdr_service.model.hpo import HPO
+from rdr_service.model.site_enums import ObsoleteStatus
+
+
+class SiteHierarchyDao(BaseDao):
+    _QUERY_PARAM_MAPPING = {
+        "awardee_id": {"col": HPO.name},
+        "organization_id": {"col": Organization.externalId},
+        "google_group": {"col": Site.googleGroup},
+    }
+
+    def __init__(self):
+        super().__init__(Site)
+
+    def get_id(self, obj):
+        pass
+
+    def handle_list_queries(self, **kwargs: Dict[str, str]) -> Response:
+        """
+        Return information on site hierarchy based on query parameters.
+
+        :param kwargs: Contains the query params extracted from the URL.
+        :return: A dictionary representing the response to be sent to the client.
+        :raises BadRequest: If a client provides an incorrect query param.
+
+        **Note:**
+        - Currently handles 3 query params: awardee_id, organization_id, google_group
+            Any query parameters not listed above will be ignored.
+        - The function returns a list of site hierarchy models that match the specified query parameters.
+        """
+        self._validate_query_params(**kwargs)
+
+        filters = [
+            self._QUERY_PARAM_MAPPING[key]["col"] == value
+            for key, value in kwargs.items()
+            if key in self._QUERY_PARAM_MAPPING
+        ]
+        with self.session() as session:
+            query_ = self.get_site_hierarchy_query(session)
+            if filters:
+                query_ = query_.filter(*filters)
+            result = query_.all()
+
+        response = {"data": [model._asdict() for model in result]}
+        return jsonify(response)
+
+    def _validate_query_params(self, **kwargs: Dict[str, str]):
+        for key, _ in kwargs.items():
+            if key not in self._QUERY_PARAM_MAPPING:
+                raise BadRequest(
+                    f"Invalid query parameter(s). "
+                    f"Supported parameters are: {', '.join(self._QUERY_PARAM_MAPPING.keys())}"
+                )
```

Comment:
Since we are not utilizing the secondary arg in items can we just just this to keys?